bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/SourceDetailsEditor.scala

```diff
@@ -0,0 +1,266 @@
+      setDistribution(libraryStarDetails.selection.item)
+      spt.notifyOfGenericUpdate()
+
+    case SelectionChanged(`libraryNonStarDetails`) =>
+      setDistribution(libraryNonStarDetails.selection.item)
+      spt.notifyOfGenericUpdate()
+
+    case SelectionChanged(`profiles`)  =>
+      setProfile(profiles.selection.item.default)
+      spt.notifyOfGenericUpdate()
+
+    case MouseClicked(_,_,_,_,_)      =>
+      if (!enabled) {
+        // first click will set profile & distribution to some default values
+        // up to that moment both values are set to None
+        setProfile(defaultProfile)
+        setDistribution(defaultDistribution)
+        spt.notifyOfGenericUpdate()
+      }
+  }
+
+  // react to any kind of target change by updating all UI elements
+  def edit(obsContext: GOption[ObsContext], spTarget: SPTarget, node: ISPNode): Unit = {
+
+    deafTo(editElements:_*)
+
+    spt = spTarget
+    isBase = if (obsContext.isDefined) obsContext.getValue.getTargets.getBase == spTarget else false
+
+    spt.getTarget.getSpatialProfile.getOrElse(defaultProfile) match {
+      case s: PointSource => profiles.selection.item = profilePanels.head;
+      case s: GaussianSource => profiles.selection.item = profilePanels(1); profilePanels(1).panel.asInstanceOf[NumericPropertySheet[GaussianSource]].edit(obsContext, spTarget, node)
+      case s: UniformSource => profiles.selection.item = profilePanels(2);
+    }
+
+    spt.getTarget.getSpectralDistribution.getOrElse(defaultDistribution) match {
+      case s: LibraryStar    => distributions.selection.item = distributionPanels.head; libraryStarDetails.selection.item = s
+      case s: LibraryNonStar => distributions.selection.item = distributionPanels(1);   libraryNonStarDetails.selection.item = s
+      case s: BlackBody      => distributions.selection.item = distributionPanels(2);   distributionPanels(2).panel.asInstanceOf[NumericPropertySheet[BlackBody]].edit(obsContext, spTarget, node)
+      case s: EmissionLine   => distributions.selection.item = distributionPanels(3);   distributionPanels(3).panel.asInstanceOf[NumericPropertySheet[EmissionLine]].edit(obsContext, spTarget, node)
+      case s: PowerLaw       => distributions.selection.item = distributionPanels(4);   distributionPanels(4).panel.asInstanceOf[NumericPropertySheet[PowerLaw]].edit(obsContext, spTarget, node)
+      case s: UserDefined    => throw new Error("not yet supported") // at a later stage we will add support for aux files user spectras
+    }
+
+    update()
+
+    listenTo(editElements:_*)
+
+  }
+
+  // show/hide UI elements as needed
+  private def update(): Unit = {
+    if (isBase) {
+      visible = true
+
+      profilePanels.foreach(_.panel.setVisible(false))
+      profiles.selection.item.panel.setVisible(true)
+
+      distributionPanels.foreach(_.panel.setVisible(false))
+      distributions.selection.item.panel.setVisible(true)
+
+      enableAll(spt.getTarget.getSpatialProfile.isDefined)
```

Comment:
There's a general issue with enabled state right now. Perhaps it would be best to just check in the version as it is now and then I can take a look at it as part of OCSADV-330.